apps/hash-api/src/graph/context-types.ts

```diff
@@ -3,11 +3,13 @@ import { DataSource } from "apollo-datasource";
 
 import { AuthenticationContext } from "../graphql/authentication-context";
 import { UploadableStorageProvider } from "../storage/storage-provider";
+import { TemporalClient } from "../temporal";
 
 export type GraphApi = GraphApiClient & DataSource;
 
 export type ImpureGraphContext<WithUpload extends boolean = false> = {
   graphApi: GraphApi;
+  temporalClient: TemporalClient | null;
```

Comment:
If this was omittable then you wouldn't need to change it everywhere... alternatively could take an approach like the `WithUpload` generic whereby the function that needs `temporalClient` is responsible for declaring that it needs it, with limits the required changes to only the functions that need it and their callers. If it's simple to revert the commit that made all the changes and do one of these things it would save future hassle